### ISSUE TITLE
How to use images with no annotations during training ?

### ISSUE THREAD

**Issue by MahmoudEwaisha** [dbolya/yolact#263]
I'm training the network on a custom dataset with only one class (similar to https://github.com/dbolya/yolact/issues/65), but i want to load all images (with and without annotations).
I did a few modifications in coco.py (COCODetection class) and augmentations.py files, so i now load and augment all images with no problems.
However, i don't know what the "pull_item" method in COCODetection should return.

You set the "masks" and "target" to None if target is zero (i.e. no annotations), but returning None will not work.

I can return a numpy array of zeros with same image size for the mask, but what should i do for the target (bounding box and label) ??

Thank you very much

**Comment by dbolya** [comment]
You'd have to make a lot of changes in the code to support images with no ground truth (in which case you'd return an empty tensor from that function). The most important changes you'd have to make would be in calculating the loss functions in MultiboxLoss, since none of the operations there support empty tensors, so you'd have to add checks around everything. I think only the confidence loss makes sense to train in that case, so basically you'd have to disable every other lose when you get an empty tensor for your GT.
